test/functional/tests/failover_standby/test_cache_activation_perf.py

```diff
@@ -0,0 +1,166 @@
+@pytest.mark.parametrize("prefill", [True, False])
+def test_cache_activation_time(prefill):
+    """
+    title: Measure standby cache activation time
+    description: Determine if the activation of a standby cache completes within
+                 a predefined time limit, on empty and prefilled devices
+    pass_criteria:
+      - The activation completes within the time limit
+    """
+    with TestRun.step(f"Setting up two cache devices, {test_drive_size} each"):
+        cache_devices = [TestRun.disks['cache_1'], TestRun.disks['cache_2']]
+        core_device = TestRun.disks['core']
+
+        # cache devices - check size and create partitions
+        for cache_dev in cache_devices:
+            check_disk_size(cache_dev)
+            cache_dev.create_partitions([test_drive_size])
+
+        # core partition creation
+        check_disk_size(core_device)
+        core_device.create_partitions([test_drive_size])
+
+    with TestRun.step("Start cache and add core"):
+        cache = casadm.start_cache(cache_devices[0].partitions[0],
+                                   force=True,
+                                   cache_mode=CacheMode.WB,
+                                   cache_line_size=cls)
+
+        core = cache.add_core(core_device.partitions[0])
+        current_cache_id = cache.cache_id
+
+    with TestRun.step("Disable seq. cutoff, cleaning policy and udev"):
+        Udev.disable()
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    if prefill is True:
+        with TestRun.step("Pre-filling the cache with dirty data"):
+
+            fill_size = test_drive_size
+            (
+                Fio()
+                .create_command()
+                .io_engine(IoEngine.libaio)
+                .block_size(Size(1, Unit.MiB))
+                .read_write(ReadWrite.write)
+                .target(core)
+                .size(fill_size)
+                .direct()
+                .num_jobs(4)
+                .run()
+            )
+
+        with TestRun.step("Check usage statistics after pre-fill"):
+            stats = cache.get_statistics(percentage_val=True)
+
+            if stats.usage_stats.dirty < dirty_threshold or \
+                    stats.usage_stats.occupancy < occupancy_threshold:
+                TestRun.LOGGER.error(f"Cache is not pre-filled correctly, "
+                                     f"dirty={stats.usage_stats.dirty}%, "
+                                     f"occupancy={stats.usage_stats.occupancy}%")
+    else:
+        TestRun.LOGGER.info("Testing on a non-prefilled cache")
+
+    with TestRun.step("Get metadata size"):
+        dmesg_out = TestRun.executor.run_expect_success("dmesg").stdout
+        md_size = dmesg.get_metadata_size_on_device(dmesg_out)
+
+    with TestRun.step("Dump metadata from the active cache device"):
+        dump_file_path = "/tmp/test_cache_activation_perf.dump"
```

Comment:
maybe there is a functionality in test framework to allocate temporary file path (like a wrapper for mktemp)?

Reply:
not relevant, as temp file was removed altogether